lib/core/src/Cardano/Wallet/Primitive/Types/UTxOIndex/Internal.hs

```diff
@@ -178,25 +163,26 @@ data UTxOIndex = UTxOIndex
 
 instance NFData UTxOIndex
 
+instance Semigroup UTxOIndex where
+    a <> b = fromSequence (toList a <> toList b)
```

Comment:
@rvl Is there any reason to add this instance of `Semigroup` for `UTxOIndex`?

If so, perhaps we can add a check to `UTxOIndexSpec` for lawfulness?

Like so:

```patch
     describe "Cardano.Wallet.Primitive.Types.UTxOIndexSpec" $ do
    
     parallel $ describe "Class instances obey laws" $ do
         testLawsMany @UTxOIndex
         [ eqLaws
+        , monoidLaws
+        , semigroupLaws
         ]  
```